BHoM_Adapter/AdapterActions/Pull.cs

```diff
@@ -42,6 +42,33 @@ public abstract partial class BHoMAdapter
         /* These methods represent Actions that the Adapter can complete. 
            They are publicly available in the UI as individual components, e.g. in Grasshopper, under BHoM/Adapters tab. */
 
+        [Description("Performs a set up for the Request input of the Pull Action.")]
+        public virtual bool SetupPullRequest(object request, out IRequest actualRequest)
+        {
+            // Always assure there is a Request. Allow to input a Type to generate a FilterRequest.
+            actualRequest = null;
+
+            if (request is Type)
+                actualRequest = BH.Engine.Data.Create.FilterRequest((Type)request, "");
+            else if (request == null)
+                actualRequest = new FilterRequest();
+            else if (typeof(IRequest).IsAssignableFrom(request.GetType()))
```

Comment:
Think the line you had worked, but for consistency with the type checking above, would change to this

```suggestion
            else if (request is IRequest)
```